src/main/java/randoop/main/GenTests.java

```diff
@@ -685,144 +764,28 @@ private void handleFlakySequenceException(AbstractGenerator explorer, SequenceEx
-   * GenInputsAbstract#progressdisplay} is true.
-   *
-   * @param files the list of {@code File} objects
-   */
-  private void logTestFiles(List<File> files) {
-    if (GenInputsAbstract.progressdisplay) {
-      System.out.println();
-      for (File f : files) {
-        System.out.println("Created file: " + f.getAbsolutePath());
-      }
-    }
-  }
-
-  /**
-   * Writes the test driver for the test classes created by the JUnit creator.
-   *
-   * @param junitPrefix the prefix of the driver class name
-   * @param junitCreator the {@link JUnitCreator} used to create test classes
-   * @param testClassNames the set of names of the generated test classes
-   * @param jfw the writer to output the Java file for the test driver
-   * @return the {@code File} for the test driver Java file
-   */
-  private File outputTestDriver(
-      String junitPrefix,
-      JUnitCreator junitCreator,
-      Set<String> testClassNames,
-      JavaFileWriter jfw) {
-    String classSource;
-    String driverName = junitPrefix;
-    if (GenInputsAbstract.junit_reflection_allowed) {
-      classSource = junitCreator.createSuiteClass(driverName, testClassNames);
-    } else {
-      driverName = junitPrefix + "Driver";
-      classSource = junitCreator.createTestDriver(driverName, testClassNames);
-    }
-    return jfw.writeClass(junit_package_name, driverName, classSource);
-  }
-
-  /**
-   * Creates the JUnit test classes for the given sequences, in AST (abstract syntax tree) form.
+   * Creates the JUnit test classes for the given sequences.
    *
    * @param junitPrefix the class name prefix
    * @param sequences the sequences for test methods of the created test classes
    * @param junitCreator the JUnit creator to create the abstract syntax trees for the test classes
    * @return mapping from a class name to the abstract syntax tree for the class
    */
-  private Map<String, CompilationUnit> getTestASTMap(
+  private LinkedHashMap<String, CompilationUnit> getTestASTMap(
       String junitPrefix, List<ExecutableSequence> sequences, JUnitCreator junitCreator) {
+
     List<List<ExecutableSequence>> sequencePartition =
         CollectionsExt.formSublists(new ArrayList<>(sequences), testsperfile);
-    Map<String, CompilationUnit> testMap = new LinkedHashMap<>();
-    String methodNamePrefix = "test";
-    String classNameFormat = junitPrefix + "%d";
+
+    LinkedHashMap<String, CompilationUnit> testMap = new LinkedHashMap<>();
     for (int i = 0; i < sequencePartition.size(); i++) {
       List<ExecutableSequence> partition = sequencePartition.get(i);
-      String testClassName = String.format(classNameFormat, i);
+      String testClassName = junitPrefix + i;
       CompilationUnit classAST =
-          junitCreator.createTestClass(testClassName, methodNamePrefix, partition);
-      testMap.put(testClassName, classAST);
+          junitCreator.createTestClass(testClassName, TEST_METHOD_NAME_PREFIX, partition);
+      if (classAST != null) {
```

Comment:
How can this happen?  If it is expected and no error, document why.  Otherwise, throw an error.